### PR TITLE
Tests: check existing username case-insensitive

### PR DESCRIPTION
**Pull request by ptamarit**
Fixes https://github.com/zenodo/ops/issues/326

:heart: Thank you for your contribution!

### Description

~~Usernames are enforced to be unique in the DB regardless of their case ([see related DB column and constraint definition](https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/alembic/eb9743315a9d_add_userprofile.py#L53-L66)).~~

~~This change applies the same validation at the registration of a user to avoid hitting the DB constraint which is then not well handled by Flask (users end up seeing an nginx 502 error).~~

This is a test-only pull request related to https://github.com/inveniosoftware/invenio-accounts/pull/476.
Here we only improve the test to make sure that `invenio-accounts` properly finds by usernames case-insensitively.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_userprofiles/forms.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def validate_username(self, field):
 
         # Check if username is already taken.
         datastore = current_app.extensions["security"].datastore
-        user = datastore.find_user(username=field.data)
+        # The `_username` DB column stores a lower-case version of the display name.
+        user = datastore.find_user(_username=field.data.lower())
```
Comment (zzacharo) [inline invenio_userprofiles/forms.py:94]: What about using the [UserProfileProxy.get_by_username()](https://github.com/inveniosoftware/invenio-userprofiles/blob/master/invenio_userprofiles/models.py#L72)
```suggestion
        user = UserProfileProxy.get_by_username(field.data)
```

That way we can avoid replicate the inner function with access to the "private" `_username`.

Comment (lnielsen) [reply invenio_userprofiles/forms.py:94]: Comment: I think this is better fixed in Invenio-Accounts on the [model itself](https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/models.py#L188-L207) using the custom [comparator on the hybrid property](https://docs.sqlalchemy.org/en/14/orm/extensions/hybrid.html#building-custom-comparators). Something along the lines of:

```
from sqlalchemy.ext.hybrid import Comparator

@username.comparator
def username(cls):
     return CaseInsensitiveComparator(cls._username)
```

This ensures that anything using the model will perform a correct database query.

Comment (ptamarit) [reply invenio_userprofiles/forms.py]: Done, I implemented this in https://github.com/inveniosoftware/invenio-accounts/pull/476

**Comment by ptamarit** [comment]
This is now a test-only pull request (see updated description).
One question: should I bump the minimum version of `invenio-accounts` from [`>=2.0.0`](https://github.com/inveniosoftware/invenio-userprofiles/blob/master/setup.cfg#L31) to `>=4.0.3`, since test would fail if using a smaller version?

**Pull request merged**